core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -419,35 +419,41 @@ class GroupCoordinator(val brokerId: Int,
     }
   }
 
-  def handleLeaveGroup(groupId: String, memberId: String, responseCallback: Errors => Unit): Unit = {
+  def handleLeaveGroup(groupId: String,
+                       memberId: String,
+                       groupInstanceId: Option[String],
+                       responseCallback: LeaveGroupResult => Unit): Unit = {
     validateGroupStatus(groupId, ApiKeys.LEAVE_GROUP).foreach { error =>
-      responseCallback(error)
+      responseCallback(leaveError(memberId, groupInstanceId, error))
       return
     }
 
     groupManager.getGroup(groupId) match {
       case None =>
-        responseCallback(Errors.UNKNOWN_MEMBER_ID)
+        responseCallback(leaveError(memberId, groupInstanceId, Errors.UNKNOWN_MEMBER_ID))
 
       case Some(group) =>
         group.inLock {
           if (group.is(Dead)) {
-            responseCallback(Errors.COORDINATOR_NOT_AVAILABLE)
+            responseCallback(leaveError(memberId, groupInstanceId, Errors.COORDINATOR_NOT_AVAILABLE))
           } else if (group.isPendingMember(memberId)) {
             // if a pending member is leaving, it needs to be removed from the pending list, heartbeat cancelled
             // and if necessary, prompt a JoinGroup completion.
             info(s"Pending member $memberId is leaving group ${group.groupId}.")
             removePendingMemberAndUpdateGroup(group, memberId)
             heartbeatPurgatory.checkAndComplete(MemberKey(group.groupId, memberId))
-            responseCallback(Errors.NONE)
-          } else if (!group.has(memberId)) {
-            responseCallback(Errors.UNKNOWN_MEMBER_ID)
+            responseCallback(leaveError(memberId, groupInstanceId, Errors.NONE))
+          } else if (!group.has(memberId) && !group.hasStaticMember(groupInstanceId)) {
+            responseCallback(leaveError(memberId, groupInstanceId, Errors.UNKNOWN_MEMBER_ID))
           } else {
-            val member = group.get(memberId)
-            removeHeartbeatForLeavingMember(group, member)
-            info(s"Member ${member.memberId} in group ${group.groupId} has left, removing it from the group")
-            removeMemberAndUpdateGroup(group, member, s"removing member $memberId on LeaveGroup")
-            responseCallback(Errors.NONE)
+              val member = if (group.hasStaticMember(groupInstanceId))
+                group.get(group.getStaticMemberId(groupInstanceId))
+              else
+                group.get(memberId)
+              removeHeartbeatForLeavingMember(group, member)
+              info(s"Member ${member.memberId} in group ${group.groupId} has left, removing it from the group")
```

Comment:
nit: if this member is a static member, better also log its instance id as well.